File: app-service-comments/src/main/java/com/github/arhor/aws/graphql/federation/comments/api/graphql/dataloader/UserRepresentationBatchLoader.java
package com.github.arhor.aws.graphql.federation.comments.api.graphql.dataloader;

import com.github.arhor.aws.graphql.federation.comments.generated.graphql.types.User;
import com.github.arhor.aws.graphql.federation.comments.service.UserRepresentationService;
import com.netflix.graphql.dgs.DgsDataLoader;
import org.springframework.beans.factory.annotation.Qualifier;

import java.util.UUID;
import java.util.concurrent.Executor;

import static org.springframework.boot.autoconfigure.task.TaskExecutionAutoConfiguration.APPLICATION_TASK_EXECUTOR_BEAN_NAME;

@DgsDataLoader(maxBatchSize = 50)
public class UserRepresentationBatchLoader extends AbstractMappedBatchLoader<UUID, User> {

    public UserRepresentationBatchLoader(
        @Qualifier(APPLICATION_TASK_EXECUTOR_BEAN_NAME) final Executor executor,
        final UserRepresentationService userService
    ) {
        super(executor, userService::findUsersRepresentationsInBatch);
    }
}

File: app-service-posts/src/main/kotlin/com/github/arhor/aws/graphql/federation/posts/api/graphql/dataloader/UserRepresentationBatchLoader.kt
package com.github.arhor.aws.graphql.federation.posts.api.graphql.dataloader

import com.github.arhor.aws.graphql.federation.posts.generated.graphql.types.User
import com.github.arhor.aws.graphql.federation.posts.service.UserRepresentationService
import com.netflix.graphql.dgs.DgsDataLoader
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.boot.autoconfigure.task.TaskExecutionAutoConfiguration.APPLICATION_TASK_EXECUTOR_BEAN_NAME
import java.util.UUID
import java.util.concurrent.Executor

@DgsDataLoader(maxBatchSize = 50)
class UserRepresentationBatchLoader(
    @Qualifier(APPLICATION_TASK_EXECUTOR_BEAN_NAME)
    executor: Executor,
    userService: UserRepresentationService,
) : AbstractMappedBatchLoader<UUID, User>(
    executor = executor,
    loaderFn = userService::findUsersRepresentationsInBatch,
)
